### PR TITLE
OEP-0060: Open Source Security Working Group

### PR DESCRIPTION
**Pull request by alangsto**
This PR introduces OEP-0060, which outlines a process for a new Open Source Security Working Group.

### CODE REVIEW

**Review by schenedx — APPROVED** [review]
I believe this is a positive step towards security management with openedx. In my experience, this format aligns to other open source repository practices. Not necessarily saying the new format will be better, but definitely invites better collaboration with the community. The increased security risk needs to be tested before we can clearly know what they are. I support this new format.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (timmc-edx) [inline oeps/processes/oep-0060-proc-sec-group.rst:114]: [nit] typo
```suggestion
vulnerable components.
```

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (timmc-edx) [inline oeps/processes/oep-0060-proc-sec-group.rst:116]: Maybe add something in here about expectations around embargo? That is, that members of the WG (and their employers) agree to keep vulnerabilities private until the coordinated disclosure occurs. This wouldn't preclude early patching unless that would require making the fix public early.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (feanil) [reply oeps/processes/oep-0060-proc-sec-group.rst:82]: I agree with this idea, we should add sending a notice that the patch is coming to everyone as a step before we make the patch public.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (feanil) [reply oeps/processes/oep-0060-proc-sec-group.rst:116]: I like this, @alangsto let's add it.

**Review by feanil — COMMENTED** [review]
Thanks for the great early feedback @timmc-edx

**Comment by timmc-edx** [comment]
I think this is looking great, I just have one more question -- how will updates be handled? Will the new working group have its own process document that it evolves independently, or will process changes be made via this OEP? (I would prefer the former in general.)

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (alangsto) [reply oeps/processes/oep-0060-proc-sec-group.rst:119]: Updated the responsibilities to include this

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (alangsto) [reply oeps/processes/oep-0060-proc-sec-group.rst:84]: Updated to include that a disclosure will be sent prior to the patch becoming public

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (alangsto) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Wording was messy on that part. The idea is that the openedx security email will be added to both the SECURITY.md and to a page on the openedx.org domain.

**Comment by feanil** [comment]
@timmc-edx most working groups have an area in the wiki and I would imagine that we would use that for the day-to-day kind of docs we would need to operate the working group.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (pshiu) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Nice catch, thanks!

**Comment by feanil** [comment]
We're not planning on making security audits a part of the initial work but I could imagine that being part of the proactive work we do.

**Review by bradenmacdonald — APPROVED** [review]

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (robrap) [inline oeps/processes/oep-0060-proc-sec-group.rst:123]: 1. Must all patches be announced ahead of time? Is there some minimum warning time that others must be given before making a fix public?
2. Maybe the meaning of "patch will become public" should be spelled out explicitly. It could mean the code becomes public in a merged commit (I'm guessing it is this). It could mean in a public PR (more on this elsewhere). It could mean when it is announced.
3. You may want a timeline for how quickly each of the following steps are implemented. Should all steps be completed within an hour (or less or more?) from merge?

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (robrap) [inline oeps/processes/oep-0060-proc-sec-group.rst:111]: 1. Is this "should" or "must"? 
2. Can a maintainer decide to use a public PR instead, and pretend that the fix isn't public until merged? Does it depend on severity? Would there be repercussions for being able to stay a maintainer?
3. If someone mistakenly creates a public fix PR, should we lay out the process of how to handle? Is it simply to close the PR and delete the branch? Should the PR subject and description be cleaned as-needed? Other?

**Review by robrap — COMMENTED** [review]
Great work. Some initial thoughts.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (Agrendalath) [inline oeps/processes/oep-0060-proc-sec-group.rst:214]:
```suggestion
* Please watch the Open edX Discourse Security Announcements topic at https://discuss.openedx.org/c/announcements/security/19. If you are logged in, select the button with a bell icon on the top right corner above the topic list and choose "Watching First Post". Discourse should send the announcements to your email that have "[Open edX discussions] [Announcements/Security]" in the subject line.
```

Nit.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (alangsto) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Updated language to better reflect expectations around security vulnerability privacy. 

As far as your second question, if a maintainer purposefully chooses to make a patch public, the process for deciding what to do would better fall under the Maintainer OEP, specifically [Removing a Maintainer](https://github.com/openedx/open-edx-proposals/blob/f43defa5cb60ac0534921bf5d4427e515dfdf0f5/oeps/processes/oep-0055-proc-project-maintainers.rst#removing-a-maintainer) if it has to go that far.

File: oeps/processes/oep-0060-proc-sec-group.rst
Comment (robrap) [reply oeps/processes/oep-0060-proc-sec-group.rst]: Looks good. Thanks!

**Review by feanil — APPROVED** [review]
The comment period has concluded and all feedback has been integrated into the OEP.  With no major issues outstanding, this OEP is approved!  🎉  🎊

**Pull request merged**
